### PR TITLE
Expand C-graphite test walker counts with different cell sizes

### LINKED ISSUE #4081: Update C molecule and solid performance tests
Adjust the number of walkers like in #4050.
Increase the number of walkers in low electron count systems while keep 16 for high electron count ones.

Note to myself. Add h5 url download link for curl before merging the updates.

### PR DESCRIPTION
**Pull request by Hyeondeok-Shin**
## Proposed changes

Close #4081. 
This PR is to expand C-graphite performance test walker counts with different cell sizes.
Test for cell size e144, e64, e16 are added for C-graphite.

## What type(s) of changes does this code introduce?

- Other (please describe): Add test

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Intel Xeon

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Review by prckent — CHANGES_REQUESTED** [review]
I think that QMC_C_MAX_SIZE and QMC_NIO_MAX_SIZE should be mentioned in our docs. The latter is overdue. Find a spot somewhere around https://qmcpack.readthedocs.io/en/develop/installation.html#performance-tests

**Review by prckent — DISMISSED** [review]
Many thanks Hyeondeok.

LGTM.

I have updated the scripts for the nightlies on sulfur and nitrogen locally. We can check the results on cdash tomorrow and see if all is in order. Once we are satisfied, please add an entry to the manual.

**Comment by prckent** [comment]
Test this please

File: tests/performance/NiO/CMakeLists.txt [hunk 1]
```diff
@@ -129,9 +129,9 @@ else()
       16
       16)
 
-  if(QMC_NIO_MAX_SIZE)
+  if(QMC_PERFORMANCE_NIO_MAX_ATOMS)
```
Comment (ye-luo) [inline tests/performance/NiO/CMakeLists.txt:132]: Could you add a trap.
```
if(QMC_NIO_MAX_SIZE)
  message(FATAL_ERROR "Variable QMC_NIO_MAX_SIZE has been renamed to QMC_PERFORMANCE_NIO_MAX_ATOMS")
endif()

**Comment by prckent** [comment]
Test this please

**Comment by prckent** [comment]
@ye-luo OK?

**Review by ye-luo — APPROVED** [review]
LGTM

**Pull request merged**